### PR TITLE
Feat/virtual protocol

### PR DESCRIPTION
**Pull request by kahwaipd**
Added virtualprotocol/ignis tool

### CODE REVIEW

**Comment by kahwaipd** [comment]
@0xArdi kindly review

File: packages/virtualprotocol/customs/ignis/ignis.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            
+            
+            return response.json()["accessToken"]
+        except (KeyError, json.JSONDecodeError) as e:
+            raise ConnectionError(f"Failed to generate access token: {e}") from e
+        
+
+def run(**kwargs) -> Tuple[Optional[str], Optional[Dict[str, Any]], Any, Any]:
+    """Run the task"""
+    api_keys = kwargs["api_keys"]['ignis'].split(":")
```
Comment (0xArdi) [inline packages/virtualprotocol/customs/ignis/ignis.py:109]: @kahwaipd where can we get such an api key for testing purposes?

**Comment by 0xArdi** [comment]
@kahwaipd the tool looks good, can you address the failing checks please?
I see that poetry.lock is not updated. Also, make sure you lock the packages before you push, you can use the following command:
```
autonomy packages lock
```

File: packages/virtualprotocol/customs/ignis/ignis.py (hunk 1 above)
Comment (kahwaipd) [reply packages/virtualprotocol/customs/ignis/ignis.py:109]: I'll share it in Telegram group.

**Comment by 0xArdi** [comment]
@kahwaipd can you please address the check failures? If you have any questions let me know.

**Comment by kahwaipd** [comment]
@0xArdi I have executed the "autonomy packages lock" , can you please check?

**Comment by 0xArdi** [comment]
@kahwaipd any chance you can resolve some of these issues?

**Comment by 0xArdi** [comment]
We are in the process of migrating the project to a [new](https://github.com/valory-xyz/mech-predict) location. 
As part of the transition, we are **closing your open pull request**.
- **No new tool PRs will be merged here**.
- This repo will only contain a bare minimum set of the dependencies. This repo will serve as a base repo for people looking to develop their own mechs and follow this as an example in the future.
- We invite you to go through our [quickstart](https://github.com/valory-xyz/mech-quickstart) on how to setup your mech and tools.

Thank you for your contributions!

**Pull request closed without merging**